lib/code_corps_web/controllers/password_reset_controller.ex

```diff
@@ -2,37 +2,31 @@ defmodule CodeCorpsWeb.PasswordResetController do
   @moduledoc false
   use CodeCorpsWeb, :controller
 
-  alias CodeCorps.{User, AuthToken}
-  alias Ecto.Changeset
+  alias CodeCorps.{AuthToken, User}
 
-  @doc"""
+  action_fallback CodeCorpsWeb.FallbackController
+
+  @doc """
   Requires a `token`, `password`, and `password_confirmation` and checks:
 
   1. The token exists in an `AuthToken` record, verified with
   `Phoenix.Token.verify`
-
   2. The `password` and `password_confirmation` match, and the auth token
   exists:
-
     - If yes, a `200` response will return the email.
     - If no, a `422` response will return the error.
   """
   def reset_password(conn, %{"token" => reset_token, "password" => _password, "password_confirmation" => _password_confirmation} = params) do
-    with %AuthToken{user: user} = auth_token <- AuthToken |> Repo.get_by(%{ value: reset_token }) |> Repo.preload(:user),
+    with %AuthToken{user: user} = auth_token <- AuthToken |> Repo.get_by(%{value: reset_token}) |> Repo.preload(:user),
          {:ok, _} <- Phoenix.Token.verify(conn, "user", reset_token, max_age: Application.get_env(:code_corps, :password_reset_timeout)),
-         {:ok, updated_user} <- user |> User.reset_password_changeset(params) |> Repo.update,
-         {:ok, _auth_token} <- auth_token |> Repo.delete,
-         {:ok, auth_token, _claims} = updated_user |> Guardian.encode_and_sign(:token)
+         {:ok, %User{} = updated_user} <- user |> User.reset_password_changeset(params) |> Repo.update(),
+         {:ok, _auth_token} <- auth_token |> Repo.delete(),
+         {:ok, auth_token, _claims} = updated_user |> CodeCorps.Guardian.encode_and_sign()
```

Comment:
This seems to have changed in ways I don't understand (i.e. not sure why we don't need `:token` as a param now).

Reply:
It looks like it just changed quite a bit. This looks like it is the correct implementation now.